### PR TITLE
Fix bugs with DOI registration status

### LINKED ISSUE #3267: DOI deposit status not recorded properly
**Describe the bug**
- The Identifiers interface for a single article does not show registration status due to a misnamed template variable
- The set logic incorrectly depletes the set that is used as the basis for object creation and logging, so database links and logs are not always properly generated

**Janeway version**
1.4.3-RC-6 commit 06de0084ff75d3ec73ecd20dca65915cf6ef2e1b

### PR DESCRIPTION
**Pull request by joemull**
Fixes #3267.

### CODE REVIEW

**Review by mauromsl — CHANGES_REQUESTED** [review]

File: src/identifiers/logic.py [hunk 1]
```diff
@@ -405,12 +405,14 @@ def send_crossref_deposit(test_mode, identifiers, journal=None):
...
-
+        journal = identifiers.copy().pop().article.journal
     error = False
 
     template = 'common/identifiers/crossref_doi_batch.xml'
-    template_context = create_crossref_doi_batch_context(journal, identifiers)
+    template_context = create_crossref_doi_batch_context(
+        journal,
+        identifiers.copy(),
+    )
```
[the remark this answers is not in the document]
Comment (joemull) [reply src/identifiers/logic.py]: Thanks for these comments--you are right. I've reworked these two functions a bit.

**Review by mauromsl — APPROVED** [review]
This bit is much clearer now, thanks @joemull

**Review by mauromsl — COMMENTED** [review]
Just one final comment in-line @joemull

File: src/identifiers/views.py [hunk 2]
```diff
@@ -190,7 +190,7 @@ def poll_doi(request, article_id, identifier_id):
         )
         identifier.crossrefstatus.update()
     else:
-        crossref_status = models.CrossrefStatus.objects.create(
+        crossref_status, created = models.CrossrefStatus.objects.get_or_create(
```
[the remark this answers is not in the document]
Comment (joemull) [reply src/identifiers/views.py:193]: Yeah, this is an attempt to handle a couple possible scenarios, but it is leaky. I've rewritten it for greater clarity to handle the three possible scenarios more cleanly and dependably, including comments.

**Review by mauromsl — APPROVED** [review]
Very clear now, thanks @joemull

**Pull request merged**